flow/project.py

```diff
@@ -281,24 +281,6 @@ def __init__(self, id, name, job, cmd, directives=None):
         # script engine later.
         keys_set_by_user = set(directives)
 
-        nranks = directives.get('nranks', 1)
-        nthreads = directives.get('omp_num_threads', 1)
-
-        if callable(nranks) or callable(nthreads):
-            def np_callable(job):
-                nr = nranks(job) if callable(nranks) else nranks
-                nt = nthreads(job) if callable(nthreads) else nthreads
-                return nr*nt
-
-            directives.setdefault('np', np_callable)
-        else:
-            directives.setdefault('np', nranks*nthreads)
-
-        directives.setdefault('ngpu', 0)
-        directives.setdefault('nranks', 0)
-        directives.setdefault('omp_num_threads', 0)
-        directives.setdefault('processor_fraction', 1)
-
         # Evaluate strings and callables for job:
         def evaluate(value):
```

Comment:
Resolve directives should also evaluate each value in the dictionary if the value is callable.